pinax/stripe/tests/test_actions.py

```diff
@@ -1379,6 +1380,37 @@ def test_sync_subscription_from_stripe_data(self):
         }
         subscriptions.sync_subscription_from_stripe_data(self.customer, subscription)
         self.assertEquals(Subscription.objects.get(stripe_id=subscription["id"]).status, "trialing")
+        subscription["discount"] = {
+            "object": "discount",
+            "coupon": {
+                "id": "35OFF",
+                "object": "coupon",
+                "amount_off": None,
+                "created": 1391694467,
+                "currency": None,
+                "duration": "repeating",
+                "duration_in_months": 3,
+                "livemode": False,
+                "max_redemptions": None,
+                "metadata": {
+                },
+                "percent_off": 25,
+                "redeem_by": None,
+                "times_redeemed": 1,
+                "valid": True
+            },
+            "customer": self.customer.stripe_id,
+            "end": 1399384361,
+            "start": 1391694761,
+            "subscription": subscription["id"]
+        }
+        with self.assertRaises(Coupon.DoesNotExist):
+            subscriptions.sync_subscription_from_stripe_data(self.customer, subscription)
```

Comment:
Not sure if this should raise, since the "coupon" is inline and could be created from there?!